### PR TITLE
Add a new ButtonAvatar component

### PR DESCRIPTION
**Pull request by AmeeMorris**
### 💬 Description
You may have noticed we've dropped the caret on the user dropdown in the app. Now that's gone we need a new way to show that our friendly avatar is clickable, so here's ButtonAvatar.

### ✅ Checklist
- [ ] Tests written
- [ ] Browser tested
- [ ] Added to documentation
- [ ] Added to storybook

### CODE REVIEW

File: lib/ButtonNew/ButtonIcon/ButtonIcon.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ function ButtonIcon({
 
   return (
     <ButtonBase className={classes} disabled={disabled} size={false} {...rest}>
-      <Icon name={name} title={iconTitle} />
+      <Icon name={name} title={iconTitle} defaultActiveColor={false} />
```
Comment (AmeeMorris) [inline lib/ButtonNew/ButtonIcon/ButtonIcon.js:37]: This fixes a bug for ButtonIcons inside a parent with an `is-active` class forcing our icon to become blue when we may not want that to happen

**Review by kyleharper — APPROVED** [review]
Noice! I've added 3 soft points for your consideration 👍

File: lib/ButtonNew/ButtonAvatar/ButtonAvatar.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+ButtonAvatar.defaultProps = {
+  url: '',
+  disabled: false,
+  className: '',
+  active: false
+};
+
+export default ButtonAvatar;
```
[the remark this answers is not in the document]
Comment (AmeeMorris) [reply lib/ButtonNew/ButtonAvatar/ButtonAvatar.js:34]: @kyleharper Why? I had a quick google and couldn't find much about it

File: lib/ButtonNew/stories/ButtonAvatar/ButtonAvatarStory.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+storiesOf('Components/Buttons', module).add('ButtonAvatar', () => {
+  const props = {
+    url: faker.image.avatar(),
+    initials: 'PP'
+  };
+  return (
+    <div>
+      <StoryItem title="ButtonAvatar" description="The button avatar component">
+        <div className="flex flex-wrap">
```
[the remark this answers is not in the document]
Comment (AmeeMorris) [reply lib/ButtonNew/stories/ButtonAvatar/ButtonAvatarStory.js:16]: This is just a bit of classic copy and pasting, its on the other button stories 🤷‍♀️

File: lib/ButtonNew/ButtonAvatar/ButtonAvatar.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyleharper) [reply lib/ButtonNew/ButtonAvatar/ButtonAvatar.js:34]: A very good read: https://blog.neufund.org/why-we-have-banned-default-exports-and-you-should-do-the-same-d51fdc2cf2ad

File: lib/ButtonNew/stories/ButtonAvatar/ButtonAvatarStory.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (altaudio) [reply lib/ButtonNew/stories/ButtonAvatar/ButtonAvatarStory.js:16]: This just contains the `<ButtonStoryItem />` components and ensures that they wrap at smaller screen sizes.

If we wanted to make it re-useable we could create `<ButtonStoryItemContianer />`, but I don't think it's a deal breaker!

File: lib/ButtonNew/ButtonAvatar/ButtonAvatar.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { string, bool } from 'prop-types';
+import { Avatar } from 'lib';
+import cx from 'classnames';
+import { ButtonBase } from '../ButtonBase';
+
+export function ButtonAvatar({
+  initials,
+  url,
```
Comment (altaudio) [inline lib/ButtonNew/ButtonAvatar/ButtonAvatar.js:9]: 💚 This comment is very green, but coming to this component for the first time I had no idea what `url` was for until I dipped into `<Avatar />` - maybe `avatarImageUrl` or something similar would be more descriptive!

File: lib/ButtonNew/ButtonAvatar/buttonAvatar.scss [hunk 5]
```diff
@@ -0,0 +1,31 @@
+.button-avatar {
+  @apply rounded-full p-0 border-transparent relative;
+  width: 34px;
+  height: 34px;
```
Comment (altaudio) [inline lib/ButtonNew/ButtonAvatar/buttonAvatar.scss:4]: 💚 If we can make these work with REM values, the avatar will scale better if the user zooms in and out.

Comment (altaudio) [reply lib/ButtonNew/ButtonAvatar/buttonAvatar.scss:4]: Ah, maybe - just realized this is styling the button, not the avatar!

**Review by altaudio — COMMENTED** [review]
Looks good to me! 👍 

Some very soft comments in there!

**Pull request merged**